lib/credentials/CredentialsManager.js

```diff
@@ -0,0 +1,103 @@
+const joi = require('@hapi/joi');
+const EventEmitter = require('events');
+
+const { authTypeAssumeRole } = require('../constants');
+const { AssumeRoleCredentials, getRoleArn } = require('./AssumeRoleCredentials');
+
+const vaultParamsJoi = joi.object({
+    id: joi.string().required(),
+    accountId: joi.string().required(),
+    stsclient: joi.object().required(),
```

Comment:
I'm not sure why a direct sts client is needed. There is a `AWS.ChainableTemporaryCredentials`  class that takes a role arn and "origin credentials" and takes care of assuming the role and automatically refreshing the temp keys/tokens when they expire (provided callers use the `get` method).

Reply:
wasn't aware of this. this would simplify what I had implemented